main.tf

```diff
@@ -13,7 +13,10 @@ locals {
   validate_kms_vars = var.kms_encryption_enabled && var.kms_key_crn == null ? tobool("When setting var.kms_encryption_enabled to true, a value must be passed for var.kms_key_crn") : true
   # tflint-ignore: terraform_unused_declarations
   validate_auth_policy = var.kms_encryption_enabled && var.skip_iam_authorization_policy == false && var.existing_kms_instance_guid == null ? tobool("When var.skip_iam_authorization_policy is set to false, and var.kms_encryption_enabled to true, a value must be passed for var.existing_kms_instance_guid in order to create the auth policy.") : true
-
+  # tflint-ignore: terraform_unused_declarations
+  validate_event_notification = var.enable_event_notification && var.existing_en_instance_crn == null ? tobool("When setting var.enable_event_notification to true, a value must be passed for var.existing_en_instance_crn") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_en_auth_policy = var.enable_event_notification && var.skip_en_iam_authorization_policy == false && var.existing_en_instance_crn == null ? tobool("When var.skip_en_iam_authorization_policy is set to false, and var.enable_event_notification to true, a value must be passed for var.existing_en_instance_crn in order to create the auth policy.") : true
```

Comment:
This validation is not needed because a value for `existing_en_instance_crn` is always needed if `enable_event_notification` is true (as per above validation), regardless of whether value is passed for `skip_en_iam_authorization_policy`